src/index.ts

```diff
@@ -39,29 +25,22 @@ y.command(
       demandOption: true,
       describe: "Dialect of database",
     },
+    includeSubdirs: {
+      type: "boolean",
+      default: false,
+      describe: "Whether to include all subfolders when resolving entity files",
+    },
   },
   async function (argv) {
     try {
       const path = argv.path;
       if (!fs.existsSync(path)) {
         throw new Error(`path ${path} does not exist`);
       }
-      const files = fs.readdirSync(path);
-      if (files.length === 0) {
-        throw new Error(`path ${path} does not contain any files`);
-      }
-      const extension = files[0].split(".").pop();
-      let sql = "";
-      switch (extension) {
-        case "ts":
-          sql = await loadEntities(argv.dialect as Dialect, [path + "/*.ts"]);
-          break;
-        case "js":
-          sql = await loadJSEntities(argv.dialect as Dialect, path);
-          break;
-        default:
-          throw new Error(`unknown file extension ${extension}`);
-      }
+      const pattern = argv.includeSubdirs
+        ? `${argv.path}/**/*.{ts,js}`
+        : `${argv.path}/*.{ts,js}`;
```

Comment:
Let's stick to the `path` flag and not add another flags. It should also work on existing setups. 
If we can't load the positions in some cases, that's fine too.